libsigner/src/http.rs

```diff
@@ -0,0 +1,241 @@
+// Copyright (C) 2013-2020 Blockstack PBC, a public benefit corporation
+// Copyright (C) 2020-2023 Stacks Open Internet Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::collections::HashMap;
+use std::io;
+use std::io::{Read, Write};
+
+use std::net::SocketAddr;
+
+use stacks_common::codec::MAX_MESSAGE_LEN;
+use stacks_common::deps_common::httparse;
+use stacks_common::util::chunked_encoding::*;
+
+use crate::error::EventError;
+use crate::error::RPCError;
+
+pub const MAX_HTTP_HEADERS: usize = 32;
+pub const MAX_HTTP_HEADER_LEN: usize = 4096;
+
+/// Decode the HTTP request payload into its headers and body.
+/// Returns (verb, path, table of headers, body_offset) on success
+pub fn decode_http_request(
+    payload: &[u8],
+) -> Result<(String, String, HashMap<String, String>, usize), EventError> {
```

Comment:
Could you make the result tuple an explicit type or at least a type alias for readability?

Reply:
Thank you :pray: